src/hotspot/share/opto/addnode.cpp

```diff
@@ -235,6 +235,51 @@ const Type *AddNode::add_of_identity( const Type *t1, const Type *t2 ) const {
   return NULL;
 }
 
+//---------------------------Bitmask helper------------------------------------
+// Assuming only And, Or and Shift (and also their replacement BitfieldInsert) are involved
+// in the expression, compute a bit mask covering all possible values of the expression.
+// Mask information can be used in
+// - Add into Or conversion (when value range masks of Add arguments do not intersect)
+// - Or(And+Shift,And+Shift) into BitfieldInsert conversion (when masks of Or arguments do not intersect)
+static juint value_range_mask(PhaseGVN *phase, Node* value) {
+  int opcode = value->Opcode();
+  if (opcode == Op_LShiftI && value->in(2)->is_Con()) {
```

Comment:
`Node::is_Con()` returns `true` for TOP node, so it's incorrect to assume the value is of type int after the check.